# STATISTICS 1, 2, 3 & 4: All 4 is in one file

In [2]:
# 2.1. Problem Statement: STATISTICS 1

# 1. You survey households in your area to find the average rent they are paying.
# Find the standard deviation from the following data:
# 
# $1550, $1700, $900, $850, $1000, $950.

In [3]:
rent=(1550,1700,900,850,1000,950)

In [6]:
average = sum(rent)/len(rent)
print("Average rent is $ ", round(average,2))

Average rent is $  1158.33


In [5]:
print(average)

1158.3333333333333


In [7]:
import numpy as np


In [17]:
SD = np.std([rent],ddof=1)
print("Standard Deviation for the rent is: $",round (SD,2))

Standard Deviation for the rent is: $ 367.99


#  2. Find the variance for the following set of data representing trees in California (heights in feet):

3, 21, 98, 203, 17, 9

In [18]:
import numpy as np
import pandas as pd

tree_ht= (3,21,98,203,17,9)
tree_ht

(3, 21, 98, 203, 17, 9)

In [26]:
tree_ht_var=np.var(tree_ht,ddof=1) # ddof is for the complete population and not a sample of a population
print("The variance is:  ", np.around(tree_ht_var,2),"ft-square")

The variance is:   6219.9 ft-square


##  Question 3.

In [27]:
# In a class on 100 students, 80 students passed in all subjects, 10 failed in one subject, 7 failed in two subjects and 
# 3 failed in three subjects. Find the probability distribution of the variable for number of subjects a student 
# from the given class has failed in.

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from matplotlib.pyplot import hist

prob_no_fail = 80/100
prob_one_fail = 10/100
prob_two_fail = 7/100
prob_three_fail = 3/100

prob_list = [prob_no_fail, prob_one_fail, prob_two_fail, prob_three_fail]

fail_pdf = pd.DataFrame(data={'No. Failed Subjects(X)':['X=0', 'X=1', 'X=2', 'X=3'],'P(X)': prob_list})
fail_pdf

No. Failed Subjects(X)  P(X)
0                    X=0  0.80
1                    X=1  0.10
2                    X=2  0.07
3                    X=3  0.03

In [3]:
# The sum of all probabilities(Cumulative Probability) should be equal to 1 in a probability distribution
fail_pdf ['P(X)'].sum() == 1  

True

# 2.2. Problem Statement: STATISTICS 2

In [9]:
import math
import statistics as st
from functools import reduce

# FORMULA TO CALCULATE BINOMIAL DISTRIBUTION IS :

P (‘k’ successes in ‘n’ trials) = C(n,k)sk(1−s)(n−k) 

s = the probability of success and 

(1-s) = the probability of failure (or complement of the event)

n = Total number of trials

K = number of specific events we want to obtain

Also nCr represents selection of r events from n, it can be written as: nCk = n!/k!(n−k)! 

# A test is conducted which is consisting of 20 MCQs (multiple choices questions) with every MCQ having its four options out of which only one is correct. Determine the probability that a person undertaking that test has answered exactly 5 questions wrong.

In [11]:
n = 20        # number of samples
s = 5
k = 15

# Probability of success 
p = 1/4
# Proability of Failure
q = 1 - p

# Percentage computation
# 20!/15!(5!) * ((1/4) ** 15) * ((3/4) ** 5)

array = [i for i in range(1,21)]
n_fact = reduce(lambda x,y : x * y, array)
pass_fact = reduce(lambda x,y : x * y, array[0:15])
fail_fact = reduce(lambda x,y : x * y, array[0:5])

# Probaility of 5 failures
prob= (n_fact/(pass_fact*fail_fact)) * (p ** k ) * (q ** s)
print ("PRoabailty to get 5 wrong is : ", prob)

PRoabailty to get 5 wrong is :  3.4264958230778575e-06


# A die marked A to E is rolled 50 times. Find the probability of getting a “D” exactly 5 times.


In [22]:
# Importing libraries
from scipy.stats import binom

In [23]:

# No. of Trails =  No. of times the die is rolled i.e. 50
n=50

# The die has to roll a “D” exactly 5 times
success = 5

# Probability of getting "D" when rolled once is =  1/5
p = 1/5
print("p: ", p)

p:  0.2


In [24]:
# PMF is the function that we can use to get the probability of exact values
print("The probability of getting a “D” exactly 5 times is : {:0.10f}".format(binom.pmf(success,n,p)))

The probability of getting a “D” exactly 5 times is : 0.0295312043


# Altrenatively 

In [12]:
n = 50
#success count
k = 5
# Failure
s = 50 -k    # 45

# Probabilty of success 
p = 1/5    # As it is A to E, there are only 5 possibilties
# Probiblity of failure
q = 1 - p

# Compute the probavilty
prob = (math.factorial(n) / (math.factorial(k) * math.factorial(s))) * (p ** k) * (q ** s)
print ("PRoabailty to get D 5 times is : ", prob)

PRoabailty to get D 5 times is :  0.029531204310524296


# Two balls are drawn at random in succession without replacement from an urn containing 4 red balls and 6 black balls. Find the probabilities of all the possible outcomes.


Total Number of balls : 10<br>
Probabilty of two reds    : RR : P(R) * P(R) : (4/10) * (3/9) &emsp; : 2/15 &emsp; :This because without replacement<br>
Probabilty of red & black : RB : P(R) * P(B) : (4/10) * (6/9) &emsp; : 4/15 &emsp; :This because without replacement<br>
Probabilty of black & red : BR : P(B) * P(R) : (6/10) * (4/9) &emsp; : 4/15 &emsp; :This because without replacement<br>
Probabilty of two blacks  : BB : P(B) * P(B) : (6/10) * (5/9) &emsp; : 1/3  &emsp;&nbsp;&nbsp; :This because without replacement<br>
<br>
<br>
Probabilty of 2 Reds &emsp;  &emsp; &emsp;  &emsp;         : 2/15        : 2/15<br>
Probabilty of 1 red and 1 black : 4/15 + 4/15 : 8/15<br>
Probabilty of 3 blacs &emsp; &emsp; &emsp;  &emsp;         : 1/3         : 1/3


In [25]:
# Importing libraries
from scipy.stats import binom

In [26]:
# Initializing values
r = 4 # No. of red balls
b = 6 # No. of black balls
t = r + b # Total number of balls

In [27]:
# There are only four possible combinations when the two balls are drawn at random: 
# (red, red),(red,black),(black, red), (black, black)

# Probability of (red, red)
prob_rr = ((r/t)*((r-1)/(t-1))) 
print("Probability of (red, red): ", prob_rr)

# Probability of (red, black)
prob_rb = ((r/t)*((b)/(t-1))) 
print("Probability of (red,black): ", prob_rb)

# Probability of (black, red)
prob_br = ((b/t)*((r)/(t-1))) 
print("Probability of (black, red): ", prob_br)

# Probability of (black, black)
prob_bb =  ((b/t)*((b-1)/(t-1)))
print("Probability of (black, black): ", prob_bb)

Probability of (red, red):  0.13333333333333333
Probability of (red,black):  0.26666666666666666
Probability of (black, red):  0.26666666666666666
Probability of (black, black):  0.3333333333333333


In [28]:
# The probabilty of all possible outcomes
prob_all = ((r/t)*((r-1)/(t-1)))  +  ((r/t)*((b)/(t-1)))  +  ((b/t)*((r)/(t-1)))   +  ((b/t)*((b-1)/(t-1)))
# OR
# prob_all = prob_rr + prob_rb + prob_br + prob_bb
prob_all

1.0

# 2.3. Problem Statement: STATISTICS 3

Blood glucose levels for obese patients have a mean of 100 with a standard deviation of 15. A researcher thinks that a diet high in raw cornstarch will have a positive effect on blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an effect or not.


Hypothesis: <br><br>
H0 : 𝜇 = 100 <br>
HA : 𝜇 > 100<br>
<br><br>
Population mean = 100 <br>
Population Standard Deviation = 15<br>
<br>
Sample size = 36<br>
mean = 108<br>
<br>
Use the Z statics as SD for population is given<br>
z = (sample mean - population mean)/ (standard Deviation/square root of sample size)<br><br>
Alpha = 0.05

In [15]:
Z_stat = (108-100)/(15/(36)**0.5); Z_stat

3.2

P value from the Z table table : 0.9993
Z-critical value = 1.645

### CONCLUSION
Clearly the Z computed value is greater than the Z-critical Value, indicating it is in the rejection area.<br>
Also the p value is greater than the Alpha value.<br>
As a result the null hypothesis will be rejected.

### In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple random sample of 100 voters are surveyed from each state. 

### What is the probability that the survey will show a greater percentage of Republican voters in the second state than in the first state?

The above is a Propotion Problem<br>
<br>
P1 = Propotion of Republicans in the First state = 0.52<br>
P2 = Propotion of Republicans in the Second state = 0.47<br>
p1 = Propotion of Democrats in the First state = 0.48<br>
p2 = Propotion of Democrats in the Secons state = 0.53<br>
<br>
Check whether the sample sizes are big. Remeber p1 = 1 - P1 and p2 = 1 - P2<br>
n1P1 = 100 * 0.52 = 52<br>
n1(1 - P1) = 100 * 0.48 = 48<br>
n1P2 = 100 * 0.47 = 47<br>
n1(1 - P2) = 100 * 0.53 = 53<br>
<br>
Clearly the sample sizes are big enough<br>
<br>
Difference in means = E(p1 - p2) = P1 - P2 = 0.52 - 0.47 = 0.05<br>
<br>
Standard Deviation Formula is :<br>
σd = sqrt{ [P1(1 - P1) / n1 ] + [ P2(1 - P2) / n2 ] } 


In [16]:
sd = (((0.52 * 0.48)/100) + ((0.47 * 0.53)/100)) ** 0.5; sd

0.07061869440877536

Compute the Z score for the same
z (p1 - p2) = (x - μ p1 - p2 ) / sd

In [17]:
z_score = (0 - 0.05) / sd; z_score

-0.7080278164104207

### The value from the Z-table for -0.071 is 0.24.

### You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard deviation is 209. How well did you score on the test compared to the average test taker?

In [18]:
Z = (1100 - 1026)/209
Z

0.35406698564593303

### Your score 0.354 from the mean

# 2.4. Problem Statement: STATISTICS 4

Is gender independent of education level? A random sample of 395 people were surveyed and each person was asked to report the highest education level they obtained. The data that resulted from the survey is summarized in the following table:<br>
<br>High School Bachelors Masters Ph.d. Total<br>
Female 60 54 46 41 201<br>
Male 40 44 53 57 194<br>
Total 100 98 99 98 395<br>
<br>
Question: Are gender and education level dependent at 5% level of significance? In other words, given the data collected above, is there a relationship between the gender of an individual and the level of education that they have obtained?<br>

In [29]:
import numpy as np
import scipy.stats as stats

In [32]:
female = [60, 54, 46, 41,201]
male = [40, 44, 53, 57, 194]

In [33]:
# Chi-squared Test
stats.chisquare(female,male)

Power_divergenceResult(statistic=17.941060964377133, pvalue=0.0012672618980429317)

In [34]:
print("Since pvalue < 0.05, we reject the null hypothesis that Male and Female categories are independent of each other")
print("Therefore, education level depends on the gender at 5% level of significance")

Since pvalue < 0.05, we reject the null hypothesis that Male and Female categories are independent of each other
Therefore, education level depends on the gender at 5% level of significance


### PROBLEM STATEMENT 2

2 Using the following data, perform a oneway analysis of variance using α=.05. Write up the results in APA format.
<br><br>
[Group1: 51, 45, 33, 45, 67]<br>
[Group2: 23, 43, 23, 43, 45]<br>
[Group3: 56, 76, 74, 87, 56]<br>

In [35]:
import numpy as np
import pandas as pd
import scipy.stats as stats

In [36]:
g1 = [51, 45, 33, 45, 67]
g2 = [23, 43, 23, 43, 45]
g3 = [56, 76, 74, 87, 56]

In [37]:
df = pd.DataFrame({"Group1": g1, "Group2": g2, "Group3": g3})
df

Group1  Group2  Group3
0      51      23      56
1      45      43      76
2      33      23      74
3      45      43      87
4      67      45      56

In [38]:
df.describe()

Group1     Group2     Group3
count   5.000000   5.000000   5.000000
mean   48.200000  35.400000  69.800000
std    12.377399  11.349009  13.535139
min    33.000000  23.000000  56.000000
25%    45.000000  23.000000  56.000000
50%    45.000000  43.000000  74.000000
75%    51.000000  43.000000  76.000000
max    67.000000  45.000000  87.000000

In [40]:
# The ANOVA test has important assumptions that must be satisfied in order for the associated p-value to be valid.

#     The samples are independent.
#     Each sample is from a normally distributed population.
#     The population standard deviations of the groups are all equal. This property is known as homoscedasticity.

# Since their standard deviations are close to each other. We can use One-Way ANOVA test

In [41]:
# One-Way ANOVA Test
stats.f_oneway(g1, g2, g3)

F_onewayResult(statistic=9.747205503009463, pvalue=0.0030597541434430556)

In [42]:
print("""Since pvalue < 0.05 (α=.05), The differences between the means are statistically significant.
Therefore, we reject the null hypothesis that they may be equal""")

Since pvalue < 0.05 (α=.05), The differences between the means are statistically significant.
Therefore, we reject the null hypothesis that they may be equal


### PROBLEM STATEMENT 3

Calculate F Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25.

In [43]:
# Calculate F Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25.
import numpy as np
import pandas as pd
import scipy.stats as stats

In [44]:
g1 = [10, 20, 30, 40, 50 ]
g2 = [5, 10, 15, 20, 25 ]

In [45]:
s1 = pd.Series(g1)
s2 = pd.Series(g2)

In [46]:
# F Test 
f = s1.var()/s2.var()
f

4.0